cantools/database/can/cpp_source.py

```diff
@@ -226,6 +225,13 @@ def _format_comment_no_tabs(comment):
         return ''
 
 
+# TODO this should evaluate outputs to integers/bools as well, just not only std::string or double
```

Comment:
What is the reason not to just do that now, is it harder for some reason?

Reply:
We need to add `SIG_VALTYPE_` attributes to every float or double in all of our DBC files in the signal_valtype_list section. See [DBC File Format Documentation page 5-6](http://read.pudn.com/downloads766/ebook/3041455/DBC_File_Format_Documentation.pdf).

> Signals with value types 'float' and 'double' have additional entries in the signal_valtype_list section. 
> > signal_extended_value_type_list = 'SIG_VALTYPE_' message_id signal_name signal_extended_value_type ';' ;  

> >    signal_extended_value_type = '0' | '1' | '2' | '3' ; (* 0=signed or
unsigned integer, 1=32-bit IEEE-float, 2=64-bit IEEE-double *) 

I could also infer this from the scale factor and min/max values. Would prefer to do the former option as it is the correct way.